### PR TITLE
Views

### PR DESCRIPTION
**Pull request by JacintaNgwu**
- Use preprocessed HTML file with embedded Ruby code.
- Use layouts and templates for shared content.

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,7 @@
-<h1>Here is a list of all users</h1>
\ No newline at end of file
+<body>
+    <% @users_all.each do |user| %>
+        <%= link_to "/users/#{user.id}" do %>
+        <%= render "partials/user_container", user: user %>
+        <% end %>
+    <% end %>
+</body>
```
Comment (AshakaE) [inline app/views/users/index.html.erb]: - [ ] Kindly check that the right model is being called with the right type, I `.each` will only work on an array type 
image

**Review by AshakaE — CHANGES_REQUESTED** [review]
Hi @JacintaNgwu , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Your code base looks neat 👏 
- The use of Github flow ✅ 

## Required Changes ♻️

- [ ] Kindly ensure your `User` model is present, I'm unable to create a user 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,9 +1,11 @@
 class UsersController < ApplicationController
   def index
     @users = User.all
+    @posts = @user.posts
```
Comment (VICTOR-CHUKWU) [inline app/controllers/users_controller.rb]: - [ ] i get this error undefined method for posts  kindly fix, It worked when I replaced with ` @user = User.first` you can try other things
Screenshot from 2023-04-21 17-00-56

File: app/views/users/index.html.erb [hunk 3]
```diff
@@ -1 +1,7 @@
-<h1>Here is a list of all users</h1>
\ No newline at end of file
+<body>
+    <% @users.each do |user| %>
+        <%= link_to "/users/#{user.id}" do %>
+        <%= render "partials/user_container", user: user %>
+        <% end %>
+    <% end %>
+</body>
```
Comment (VICTOR-CHUKWU) [inline app/views/users/index.html.erb]: - [ ] kindly make sure your design match the given wire frame
your design :point_down: 
Screenshot from 2023-04-21 17-06-58
given wire frame :point_down: 
Screenshot from 2023-04-21 17-06-41

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -1 +1,20 @@
...
+            <p><%= truncate(post.text, length: 110, omission: " . . .") %></p>
+            <div class="like-and-comment">
+              <p>Comments: <%= post.comment_counter || 0 %>, Likes: <%= post.like_counter || 0 %></p>
+            </div>
+              <% end %>
+        </article>
+        <%= link_to "/users/#{@user.id}/posts" do %>
+    <button>See all posts</button>
+    <% end %>
+</body>
```
Comment (VICTOR-CHUKWU) [inline app/views/users/show.html.erb]: - [ ] there is an error when I load user show page, kindly change `comment_counter` to `comments_counter` and also same for `like_counter` to `likes_counter`
Screenshot from 2023-04-21 17-09-14
- [ ] also your design is not matching the wireframe, kindly follow design
your design :point_down: 
Screenshot from 2023-04-21 17-12-12

File: app/views/posts/index.html.erb [hunk 5]
```diff
@@ -1 +1,12 @@
...
+    <% @posts.each do |post| %>
+    <h3>Post Title: <%= @posts.title %></h3>
+    <p><%= @posts.text %></p>
+    <div class="block">
+        <p>Comments: <%= @posts.comments_counter %></p>
+        <p>Likes: <%= @posts.likes_counter %></p>
+    </div>
+    <% end %>
+    <button>Pagination</button>
+</body>
```
Comment (VICTOR-CHUKWU) [inline app/views/posts/index.html.erb]: - [ ] there is an error in your post index, kindly fix this
Screenshot from 2023-04-21 17-14-02

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @JacintaNgwu , 
## STATUS: CHANGES REQUIRED :recycle: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb [hunk 6]
```diff
@@ -1 +1,11 @@
-<h1>Here is a list of all users</h1>
\ No newline at end of file
+<section class= "all">
+  <% @users.each do |user| %>
+    <div class="user_container">
+   <img class="pic" src="<%= user.photo %>" alt="profile_pic">
+   <div class=>
+   <%= link_to user.name, user_path(user)%>
+  <p class="post-count">Number of posts: <%= user.posts_counter || 0 %></p>
+</div>
```
Comment (BenMukebo) [inline app/views/users/index.html.erb]: - [ ] Kindly as requested by the previous reviewer, the user index view page should match the given [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_root_users.png) However your design does not match the wireframe👇🏻 Kindly make sure that the index view page matches the given the user index view page should match the given [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_root_users.png).

Your Implementation | Original Design
:---:|:---:
image | image


- Do the same for your User show page.

image

File: app/views/posts/index.html.erb [hunk 7]
```diff
@@ -1 +1,15 @@
...
+<% @posts.each do |post| %>
+
+<p><%= link_to post.title, user_post_path(@user,post)%></p>
+<p><%= post.text%></p>
+<p>comment:<%=post.comments_counter %></p>
+<p>likes:<%=post.likes_counter%></p>
+<h3> comments </h3>
+ <%post.comments.each do |comment|%>
+ <p><%=comment.author.name %>: <%=comment.text %></p>
+ <%end%>
```
Comment (BenMukebo) [inline app/views/posts/index.html.erb]: - [ ] You should pay more attention to the function as styling is nice to have the **See all Post** links open the user's post index as displayed on the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_all_posts.png) However your design does not match the wireframe👇🏻 Kindly make sure that the index view page matches the given  [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_all_posts.png).

Your Implementation | Original Design
:---:|:---:
image | image
 
- Do the same for your Post show page.

image

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @JacintaNgwu, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️


 ## Required Changes ♻️
_Check the comments under the review._


You have submitted a project, that Linters check are not running in the GitHub/action
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/show.html.erb [hunk 8]
```diff
@@ -1 +1,25 @@
...
+    <div class="counts">
+    <p class="likes-count"> <%= d.likes_counter%></p>, 
+    <p class="comments-count"> <%= d.comments_counter%></p>
+    </div>
+  </div>
+<%end%>
+</div>
+<button class="see_all">
+<%= link_to "see all posts", user_posts_path(@user)%>
+</button>
```
Comment (danmainah) [inline app/views/users/show.html.erb]: - [ ] @JacintaNgwu great work implementing the changes requested, however, you forgot to add the post counter title before the counter, the same case for the likes and comments counters 
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @JacintaNgwu , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — APPROVED** [review]
Hi @JacintaNgwu 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested chnages!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**